src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -250,7 +251,20 @@ def continue_upgrade(self) -> bool:
         :return:
         """
         if self.upgrade_state and not self.upgrade_state.paused:
-            self._do_upgrade()
+            try:
+                self._do_upgrade()
+            except (OrchestratorError, ImportError, NotImplementedError) as e:
+                # these exception types will be handled by orchestrator
+                raise e
```

Comment:
serve loop only handles `OrchestratorError` ..?

https://github.com/ceph/ceph/blob/cbc7f727842f93130fcfbbad64de0959ee4a02d3/src/pybind/mgr/cephadm/serve.py#L95

Reply:
:+1:  I was basing this off of what we catch in handle_exception in _interface.py but it looks like that's only for the synchronous part of the cli commands. I'll fix it.